bpfd-operator/api/v1alpha1/ebpfprogram_types.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EbpfProgramSpec defines the desired state of EbpfProgram
+type EbpfProgramSpec struct {
+	// ProgramMap is is a map with Keys: UUIDs, Values: AttachPoint
+	ProgramMap map[string]EbpfProgramAttachPoint `json:"programmap"`
+
+	// MapPins is the location on the filesystem of any pinned bpfmaps used by the program
+	// It is represented as a map with Keys: UUID, Value: Absolute path on host filesystem
+	MapPinPoints map[string]string `json:"mappinpoints"`
```

Comment:
You did this right, but looks funny to me. Keep reading it "mapping points" where you forgot the 'g'.

Reply:
haha true should I change it to just `Maps`?

Reply:
`maps` or `bpfmaps`? The way you did was fine too, just an observation.

Reply:
Probably just `maps` since it's already in an ebpfProgram object so it will read ebpfProgram.Maps

Reply:
+1